### PR TITLE
generic_log_worker: Add dedup cache validation and unit tests

### PR DESCRIPTION
**Pull request by kcxlee**
Adds validation logic to prevent dedup cache corruption when head/tail indices become invalid. Introduces `validate_head_tail()` to detect and correct corrupted states (head > tail or delta > max_batches), and `compute_cache_keys_to_load()` to safely bound the number of keys loaded from storage.

Adds metrics for dedup cache size and load errors. Updates cache loading to log detailed state information and gracefully handle failures without

### CODE REVIEW

File: crates/generic_log_worker/src/lib.rs [hunk 1]
```diff
@@ -707,3 +769,251 @@ impl ObjectBackend for CachedRoObjectBucket {
...
+        let result = validate_head_tail(0, 4_000_000, 128, "test");
+        assert_eq!(result.head, 4_000_000 - 128); // head moved up
+        assert_eq!(result.tail, 4_000_000);
+        assert!(result.head_corrected);
+        assert!(result.tail - result.head <= 128);
+    }
+
+    #[test]
+    fn test_validate_head_tail_delta_exactly_max() {
+        // Edge case: delta exactly equals max_batches
```
[the remark this answers is not in the document]
Comment (kcxlee) [reply crates/generic_log_worker/src/lib.rs]: Just in case, someone could accidentally change > to >= and break the valid case where the cache is exactly full. The test detects and enforces that 128 batches is the intended maximum capacity, not 127.
